src/parser/warrior/fury/modules/spells/MissedRampage.js

```diff
@@ -2,57 +2,88 @@ import React from 'react';
 
 import Analyzer from 'parser/core/Analyzer';
 
+import Events from 'parser/core/Events';
+
 import SPELLS from 'common/SPELLS';
 import RESOURCE_TYPES from 'game/RESOURCE_TYPES';
+import { SELECTED_PLAYER } from 'parser/core/EventFilter';
 
 import SpellLink from 'common/SpellLink';
 
-const GENERATORS = [
-  SPELLS.RAGING_BLOW.id,
-  SPELLS.BLOODTHIRST.id,
-  SPELLS.EXECUTE.id,
+/*  Example log:
+ *  https://www.warcraftlogs.com/reports/KhynM7v96cZkTBdg#fight=6&type=damage-done&source=78
+ */
+
+const rageGenerators = [
+  SPELLS.RAGING_BLOW,
+  SPELLS.BLOODTHIRST,
+  SPELLS.EXECUTE_FURY,
+  SPELLS.WHIRLWIND_FURY,
+  SPELLS.FURIOUS_SLASH_TALENT,
+  SPELLS.SIEGEBREAKER_TALENT,
+  SPELLS.DRAGON_ROAR_TALENT,
+  SPELLS.BLADESTORM_TALENT,
 ];
 
 class MissedRampage extends Analyzer {
   missedRampages = 0;
+  hasFB = false;
+
+  constructor(...args) {
+    super(...args);
 
-  isGenerator(spellId) {
-    return GENERATORS.includes(spellId);
+    this.active = true;
+    this.hasFB = this.selectedCombatant.hasTalent(SPELLS.FROTHING_BERSERKER_TALENT.id);
+    
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(rageGenerators), this.onPlayerCast);
   }
 
-  on_byPlayer_cast(event) {
-    const resource = event.classResources && event.classResources.find(classResources => classResources.type === RESOURCE_TYPES.RAGE.id);
-    if (!resource) {
+  onPlayerCast(event) {
+    const rage = event.classResources && event.classResources.find(classResources => classResources.type === RESOURCE_TYPES.RAGE.id);
+    if (!rage) {
       return;
     }
-    const rage = Math.floor(resource.amount/10);
-    if (rage >= 90 && this.isGenerator(event.ability.guid)) {
+
+    if (rage >= 90) {
       this.missedRampages += 1;
     }
   }
 
   get suggestionThresholds() {
-    return {
-      actual: this.missedRampages,
-      isGreaterThan: {
-        minor: 0,
-        average: 5,
-        major: 10,
-      },
-      style: 'number',
-    };
+    if (this.hasFB) {
+      return {
+        actual: this.missedRampages,
+        isGreaterThan: {
+          minor: 0,
+          average: 0,
+          major: 0,
+        },
+        style: 'number',
+      };
+    } else {
+      return {
+        actual: this.missedRampages,
+        isGreaterThan: {
+          minor: 0,
+          average: 5,
+          major: 10,
+        },
+        style: 'number',
+      };
+    }
   }
 
   suggestions(when) {
     when(this.suggestionThresholds).addSuggestion((suggest, actual, recommended) => {
       return suggest(
         <>
-          There were {actual} times when you cast another ability when you should have cast <SpellLink id={SPELLS.RAMPAGE.id} />.
+          There were {actual} times you casted a rage generating ability when you should have cast <SpellLink id={SPELLS.RAMPAGE.id} />.
```

Comment:
I would also include information on why they should have cast Rampage instead. This will help players that are new to the spec to understand why they are doing certain things or why it is a mistake to not do so.